### PR TITLE
feat(permissions): handle mutliple permissions cache

### PR DESCRIPTION
**Pull request by VincentMolinie**
## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by arnaudbesnier** [comment]
Task linked: [CU-d6wd9d Stripe #4 - Add optional global key to PermissionsGetter](https://app.clickup.com/t/d6wd9d)

File: src/services/permissions-getter.js [hunk 1]
```diff
@@ -19,38 +19,52 @@ class PermissionsGetter {
     return this.configStore.lianaOptions.envSecret;
   }
 
-  cleanCache() {
-    this.permissions = {};
+  cleanCache({ storePermissionsInto } = {}) {
```
Comment (larcin) [inline src/services/permissions-getter.js:22]: [MISC] What does `storePermissionsInto` mean. It makes no sense

File: src/middlewares/permissions.js [hunk 2]
```diff
@@ -51,13 +52,18 @@ class PermissionMiddlewareCreator {
...
 
-      return this.permissionsChecker
-        .checkPermissions(renderingId, this.collectionName, permissionName, permissionInfos)
-        .then(next)
-        .catch((error) => {
-          this.logger.error(error.message);
-          next(httpError(403));
-        });
+      const storePermissionsInto = this.configStore.lianaOptions.multiplePermissionsCache
+        ? this.configStore.lianaOptions.multiplePermissionsCache.getStorePermissionsIntoKey(request)
```
Comment (larcin) [inline src/middlewares/permissions.js:56]: [MISC] Same here, `getStorePermissionsIntoKey` does not bear that much sense

**Review by larcin — CHANGES_REQUESTED** [review]

File: src/services/permissions-getter.js (hunk 1 above)
Comment (larcin) [reply src/services/permissions-getter.js:22]: Something like `permissionsCacheKey` or even `environmentId` would be better suited

**Review by larcin — APPROVED** [review]
CC still unhappy

**Review by larcin — APPROVED** [review]

**Pull request merged**